python/paddle/fluid/dygraph/dygraph_to_static/convert_operators.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from paddle.fluid.framework import Variable
+from paddle.fluid.layers import control_flow, logical_and, logical_or, logical_not
+from paddle.fluid.dygraph.dygraph_to_static.variable_trans_func import to_static_variable
+
+
+def convert_while_loop(cond, body, loop_vars):
+    """
+    A function representation of a Python ``while`` statement.
+
+    Args:
+        cond(Callable): A callable object that returns a boolean variable to control whether to  execute the loop body.  It takes  ``loop_vars`` as arguments.
+        body(Callable): A callable object that returns a tuple or list of variables with the same arguments ``loops_vars`` as ``cond`` .
+        loop_vars(list|tuple): A list or tuple of variables passed to ``cond`` and ``body`` .
+
+    Returns:
+        A list or tuple of variables which returned by ``body`` .
+    """
+
+    pred = cond(*loop_vars)
```

Comment:
Great! This convert_call writing is clearer than I thought.

But there is a potential very very tiny issue:

At line 33, we calculate cond(*loop_vars) once and in  _run_py_while or _run_paddle_while_loop we calculate it again.
1. It may be slower if cond is very expensive.
2. If loop_vars is changed during cond callable, then it causes bug.

But both above issues won't happen now because usually cond is just O(1) and current logical_and/logical_not/... doesn't change the loop_vars.

Since CI has passed, and I notice that you commented some test cases for `if` case, you can change in the future PR. I will approve this PR now.

Reply:
Thanks!  Considering that I will change it in the future PR.